### PR TITLE
Dispatcher and Lookup Batching

### PR DESCRIPTION
**Pull request by KaustubhShamshery**
## Description

A lookup dispatcher which mainly has these two new features
* Lookup now collects txns in batches and sends them to the nodes
* A dispatcher (separate node) sends these txns to lookup

#509 

## Review Suggestion
Diff: [W 509](https://github.com/Zilliqa/Zilliqa/compare/feature/dsmicroblock...Kaustubh/LookupDispatcher)

### Implementation

- [x] **remove magic number**
- [ ] **delete already sent txns**
- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test
- [ ] small-scale cloud test

### CODE REVIEW

File: src/libLookup/Lookup.cpp [hunk 1]
```diff
@@ -96,6 +96,90 @@ void Lookup::SetLookupNodes()
...
+
+bool Lookup::GenTxnToSend(size_t n, map<uint32_t, vector<unsigned char>>& mp,
+                          uint32_t nShard)
+{
+    vector<unsigned char> txns;
+
+    for (auto& privKeyHexStr : GENESIS_KEYS)
+    {
+        auto privKeyBytes{DataConversion::HexStrToUint8Vec(privKeyHexStr)};
+        auto privKey = PrivKey{privKeyBytes, 0};
```
[the remark this answers is not in the document]
Comment (KaustubhShamshery) [reply src/libLookup/Lookup.cpp]: this function is just cut-pasted from node.cpp

File: src/libUtils/GetTxnFromFile.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+public:
+    //clears vec
+    static bool GetFromFile(Address addr, unsigned int startNum,
+                            unsigned int totalNum, vector<unsigned char>& vec)
+    {
+        fstream file;
+        vec.clear();
+
+        file.open("/home/kaustubh/Documents/LookupTxn/Zilliqa/txns/"
+                      + addr.hex() + ".zil",
```
Comment (ShengguangXiao) [inline src/libUtils/GetTxnFromFile.h]: Why here has a hardcoded file path?

Comment (KaustubhShamshery) [reply src/libUtils/GetTxnFromFile.h]: Mistake, fill fix asap

File: constants.xml [hunk 3]
```diff
@@ -34,6 +34,8 @@
         <BROADCAST_INTERVAL>60</BROADCAST_INTERVAL>
         <BROADCAST_EXPIRY>600</BROADCAST_EXPIRY>
         <TX_DISTRIBUTE_TIME_IN_MS>10000</TX_DISTRIBUTE_TIME_IN_MS>
+		<NUM_NODES_TO_SEND_LOOKUP>3</NUM_NODES_TO_SEND_LOOKUP>
+		<NUM_TXN_TO_SEND_PER_ACCOUNT>4000</NUM_TXN_TO_SEND_PER_ACCOUNT>
```
[the remark this answers is not in the document]
Comment (KaustubhShamshery) [reply constants.xml]: thanx, noted

**Comment by jiayaoqijia** [comment]
Please add description for your PR.

File: src/libNode/MicroBlockProcessing.cpp [hunk 4]
```diff
@@ -372,24 +403,25 @@ bool Node::ComposeMicroBlock()
...
         numTxs = processedTransactions.size();
-        tranHashes.resize(numTxs);
-        for (const auto& tx : processedTransactions)
+        if (numTxs != m_TxnOrder.size())
         {
-            const auto& txid = tx.first.asArray();
-            copy(txid.begin(), txid.end(),
-                 tranHashes.at(index).asArray().begin());
-            index++;
+            LOG_GENERAL(FATAL,
```
Comment (ansnunez) [inline src/libNode/MicroBlockProcessing.cpp]: I'm thinking if we should make it FATAL, or attempt to rejoin instead

**Comment by ansnunez** [comment]
Work done in this PR has been integrated into #551

**Pull request closed without merging**